src/map.c

```diff
@@ -443,9 +446,17 @@ int map_save_icemap(map_t *map, const char *fname)
 	int len;
 	char *buf = map_serialise_icemap(map, &len);
 
+	if (buf == NULL)
+	{
+		error_perror("map_save_icemap: could not serialise map");
+		return 1;
```

Comment:
File left hanging open.

You should move the serialisation above the open step really, and also remember to free the buffer if opening the file fails, otherwise you'll end up with a blank file if serialisation fails.